src/IO/ReadBufferFromEncryptedFile.cpp

```diff
@@ -156,11 +156,8 @@ bool ReadBufferFromEncryptedFile::nextImpl()
 
     /// Read up to the size of `encrypted_buffer`.
     size_t count = 0;
-    while (bytes_to_read && !in->eof())
-    {
-        count += in->read(encrypted_buffer.data() + count, bytes_to_read);
-        bytes_to_read -= count;
```

Comment:
The problem was here. It subtracted the number of read bytes from `bytes_to_read` multiple times and as a result the statement `working_buffer.resize(count)` below could get a wrong `count` beyond the memory allocated for the internal buffer.